lib/api/server.py

```diff
@@ -67,25 +71,22 @@ def stop(self):
     def disable_logging(self):
         logging.getLogger('werkzeug').setLevel(logging.ERROR)
 
+    # https://github.com/aluo-x/Learning_Neural_Acoustic_Fields/blob/master/train.py
+    # https://github.com/pytorch/pytorch/issues/71029
+    def find_free_port(self):
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+            s.bind(('localhost', 0))
```

Comment:
You know, looking back on it, I have no idea why I felt the need to search for ports with a loop.  This works fine.  Weird.